doc/operations/platforms/snp.rst

```diff
@@ -74,17 +73,112 @@ Governance Proposals
 
 The following governance proposals can be issued to add/remove these trusted values, e.g. when upgrading the service (see :doc:`/operations/code_upgrade`):
 
-- ``add_snp_host_data``/``remove_snp_host_data``: To add/remove a trusted security policy, e.g. when adding a new trusted container image as part of the code upgrade procedure. 
-- ``add_snp_uvm_endorsement``/``add_snp_uvm_endorsement``: To add remove a trusted UVM endorsement (Azure deployment only).
+- ``add_snp_host_data``/``remove_snp_host_data``: To add/remove a trusted security policy, e.g. when adding a new trusted container image as part of the code upgrade procedure.
+- ``add_snp_uvm_endorsement``/``remove_snp_uvm_endorsement``: To add/remove a trusted UVM endorsement (Azure deployment only).
 - ``add_snp_measurement``/``remove_snp_measurement``: To add/remove a trusted measurement.
 - ``set_snp_minimum_tcb_version_hex``/``remove_snp_minimum_tcb_version``: To add/remove a minimum trusted TCB version.
   - ``set_snp_minimum_tcb_version`` was deprecated in CCF 6.0.9 and replaced by ``set_snp_minimum_tcb_version_hex``.
 
+Code update
+~~~~~~~~~~~
+
+Check :doc:`/operations/code_upgrade` first.
+
+* Use ``add_snp_host_data`` and ``add_snp_measurement`` for the new code version.
+* After adding new nodes and retiring old ones, use ``remove_snp_host_data`` and ``remove_snp_measurement`` to remove the old values.
+
+Examples:
+
+.. code-block:: json
+
+  {
+    "actions": [
+      {
+        "name": "add_snp_host_data",
+        "args": {
+          "host_data": "hex-encoded SHA256 digest of the security policy",
+          "security_policy": "security policy text string"
+        }
+      }
+    ]
+  }
+
+  {
+    "actions": [
+      {
+        "name": "add_snp_measurement",
+        "args": {
+          "measurement": "5feee30d6d7e1a29f403d70a4198237ddfb13051a2d6976439487c609388ed7f98189887920ab2fa0096903a0c23fca1"
+        }
+      }
+    ]
+  }
+
+  {
+    "actions": [
+      {
+        "name": "remove_snp_host_data",
+        "args": {
+          "host_data": "hex-encoded SHA256 digest of the security policy"
+        }
+      }
+    ]
+  }
+
+  {
+    "actions": [
+      {
+        "name": "remove_snp_measurement",
+        "args": {
+          "measurement": "5feee30d6d7e1a29f403d70a4198237ddfb13051a2d6976439487c609388ed7f98189887920ab2fa0096903a0c23fca1"
+        }
+      }
+    ]
+  }
+
+Updating UVM endorsements
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To update the UVM endorsements, for instance, to rev up the SVN version, use ``add_snp_uvm_endorsement`` and ``remove_snp_uvm_endorsement`` governance actions:
+* If you're performing a code upgrade, and the old measurement can't be verified with the new UVM endorsement, you must add the endorsements first.
```

Comment:
This a bit misleading, CCF code upgrades are generally de-correlated from UVM endorsement updates. Those take place when the platform (C-ACI) is updated, which may or may not coincide with a release of the CCF-based application, but probably does not.